executor/show.go

```diff
@@ -305,13 +305,15 @@ func (v *visibleChecker) Leave(in ast.Node) (out ast.Node, ok bool) {
 }
 
 func (e *ShowExec) fetchShowBind() error {
-	var bindRecords []*bindinfo.BindRecord
+	var tmp []*bindinfo.BindRecord
 	if !e.GlobalScope {
 		handle := e.ctx.Value(bindinfo.SessionBindInfoKeyType).(*bindinfo.SessionHandle)
-		bindRecords = handle.GetAllBindRecord()
+		tmp = handle.GetAllBindRecord()
 	} else {
-		bindRecords = domain.GetDomain(e.ctx).BindHandle().GetAllBindRecord()
+		tmp = domain.GetDomain(e.ctx).BindHandle().GetAllBindRecord()
 	}
+	bindRecords := make([]*bindinfo.BindRecord, len(tmp))
```

Comment:
just glimpsed at the issue, should we clone the BindRecord rather than copy the pointer of it?

Reply:
the main problem is that we parallelly use this slice leading to data race. so that I copy the new slice to sort. If we parallelly use the ```BindRecord```. Cloning the BindRecord will be a good idea.

Reply:
`GetAllBindRecord` creates a new slice and return it to caller, no other goroutine will access the slice simultaneously. I think the race is caused by accessing the elements of slice? so only copy the pointer is not enough

Reply:
You are right. I creaet ```slice.Copy``` to deep copy a slice.